pyro/infer/autoguide/guides.py

```diff
@@ -1419,20 +1419,25 @@ def get_deltas(self, save_params=None):
                 aux_value = aux_value + dep(aux_values[upstream])
             aux_values[name] = aux_value
 
-            # Shift by loc, reshape, and transform to constrained space.
-            unconstrained = aux_value + loc
-            shape = self._unconstrained_shapes[name]
-            if torch._C._get_tracing_state() or unconstrained.shape != shape:
-                sample_shape = unconstrained.shape[:-1]
-                unconstrained = unconstrained.reshape(sample_shape + shape)
+            # Shift by loc and reshape.
+            batch_shape = torch.broadcast_shapes(
+                aux_value.shape[:-1], self._batch_shapes[name]
+            )
```

Comment:
It is not clear to me the relationship of `aux_value.shape[:-1]` and `fn.batch_shape`. Could you elaborate on this a bit? I feel that you need `batch_shape = aux_value.shape[:-1] + self._batch_shapes[name]` but I'm not sure.

Reply:
- `self._batch_shapes[name]` is the batch shape of the site under the original model with original plates.
- `aux_value` has `event_dim = 1` so its batch shape is `aux_value.shape[:-1]`. This is the shape of the site with all original plates stripped away (converted to event dims) but possibly other plates added, e.g. `vectorize_particles`.

This computation combines the original plates with the new inference-specific plates.  Note we need to broadcast rather than concatenate because both batch shapes are aligned at the batch|event boundary.